server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -646,6 +646,30 @@ public Set<String> resolveExpressions(ClusterState state, String... expressions)
         }
     }
 
+    /**
+     * Resolve the expression to the set of indices, aliases, and datastreams that the expression matches.
+     */
+    public Set<String> resolveExpressions(ClusterState state, IndicesOptions indicesOptions, String... expressions) {
+        Context context = new Context(
+            state,
+            indicesOptions,
+            true,
+            false,
+            true,
+            true,
```

Comment:
This both includes (line 658) and preserves (line 659) datastreams. Datastreams are usually not preserved, unless the API specifically deals with datastreams, or, in this case, the Resolve index API shows the names with the datastreams covered by an expression.

Reply:
There seems to be a risk that a future refactor could incorrectly merge this new method with the existing `Set<String> resolveExpressions(ClusterState state, String... expressions)` where the difference is not about `preserveDataStreams`. Could we be pro-active and complete the merge in this PR? That is, change the signature of the method to 
```
Set<String> resolveExpressions(ClusterState state, IndicesOptions indicesOptions, boolean preserveDataStreams, String... expressions)
```
and have the existing method delegate to it?

Reply:
Thanks for the suggestion, pushed [b4518b4](https://github.com/elastic/elasticsearch/pull/92820/commits/b4518b48c569dc69e5e9785cb3208a77a667ca38)